openedx/core/djangoapps/course_views/course_views.py

```diff
@@ -0,0 +1,176 @@
+"""
+Tabs for courseware.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+from xmodule.tabs import CourseTab, link_reverse_func
+
+_ = lambda text: text
+
+
+# Stevedore extension point namespaces
+COURSE_VIEW_TYPE_NAMESPACE = 'openedx.course_view_type'
+
+
+class CourseViewType(object):
+    """
+    Base class of all course view type plugins.
```

Comment:
Ok, I think I understand the distinction you're making between the `CourseView` and the `CourseTab` (would removing `Type` help clarify?). It might be worth expanding on the expectation that `CourseViews` might generate more than just particular tabs in the webapp (and perhaps to remove explicit references to tabs in the documentation/comments here, just to reinforce the distinction).